python/ql/test/query-tests/Security/CWE-327/ssl_fluent.py

```diff
@@ -0,0 +1,211 @@
+        with context.wrap_socket(sock, server_hostname=hostname) as ssock:
+            print(ssock.version())
+
+
+def create_relaxed_context():
+    return ssl.SSLContext(ssl.PROTOCOL_SSLv23)
+
+def create_secure_context():
+    context = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
+    context.options |= ssl.OP_NO_TLSv1 | ssl.OP_NO_TLSv1_1
+    return context
+
+def create_connection(context):
+    with socket.create_connection(('www.python.org', 443)) as sock:
+        with context.wrap_socket(sock, server_hostname=hostname) as ssock:
+            print(ssock.version())
+
+def test_delegated_context_unsafe():
+    context = create_relaxed_context()
+    with socket.create_connection(('www.python.org', 443)) as sock:
+        with context.wrap_socket(sock, server_hostname=hostname) as ssock:
+            print(ssock.version())
+
+def test_delegated_context_safe():
+    context = create_secure_context()
+    with socket.create_connection(('www.python.org', 443)) as sock:
+        with context.wrap_socket(sock, server_hostname=hostname) as ssock:
+            print(ssock.version())
+
+def test_delegated_context_made_safe():
+    context = create_relaxed_context()
+    context.options |= ssl.OP_NO_TLSv1 | ssl.OP_NO_TLSv1_1
+    with socket.create_connection(('www.python.org', 443)) as sock:
+        with context.wrap_socket(sock, server_hostname=hostname) as ssock:
+            print(ssock.version())
+
+def test_delegated_context_made_unsafe():
+    context = create_secure_context()
+    context.options &= ~ssl.OP_NO_TLSv1_1
+    with socket.create_connection(('www.python.org', 443)) as sock:
+        with context.wrap_socket(sock, server_hostname=hostname) as ssock:
+            print(ssock.version())
+
+def test_delegated_connection_unsafe():
+    context = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
+    create_connection(context)
+
+def test_delegated_connection_safe():
+    context = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
+    context.options |= ssl.OP_NO_TLSv1 | ssl.OP_NO_TLSv1_1
+    create_connection(context)
+
+def test_delegated_connection_made_safe():
+    context = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
+    context.options |= ssl.OP_NO_TLSv1 | ssl.OP_NO_TLSv1_1
+    create_connection(context)
+
+def test_delegated_connection_made_unsafe():
+    context = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
+    context.options |= ssl.OP_NO_TLSv1 | ssl.OP_NO_TLSv1_1
+    context.options &= ~ssl.OP_NO_TLSv1_1
+    create_connection(context)
+
+def test_delegated_unsafe():
+    context = create_relaxed_context()
+    create_connection(context)
+
+def test_delegated_safe():
+    context = create_secure_context()
+    create_connection(context)
+
+def test_delegated_made_safe():
+    context = create_relaxed_context()
+    context.options |= ssl.OP_NO_TLSv1 | ssl.OP_NO_TLSv1_1
+    create_connection(context)
+
+def test_delegated_made_unsafe():
+    context = create_secure_context()
+    context.options &= ~ssl.OP_NO_TLSv1_1
+    create_connection(context)
```

Comment:
And that the path logic (the source-sink-sanitizer-setup) is appropriate. It may not be that important, since we a re not chaining configurations, though..

Reply:
Ok :+1: